interfaces/apparmor/template.go

```diff
@@ -386,6 +386,15 @@ var defaultTemplate = []byte(`
   /sys/class/ r,
   /sys/class/**/ r,
 
+  # Workaround https://launchpad.net/bugs/359338 until upstream handles
+  # stacked filesystems generally.
+  # encrypted ~/.Private and old-style encrypted $HOME
+  @{HOME}/.Private/ r,
+  @{HOME}/.Private/** mrixwlk,
+  # new-style encrypted $HOME
+  @{HOMEDIRS}/.ecryptfs/*/.Private/ r,
+  @{HOMEDIRS}/.ecryptfs/*/.Private/** mrixwlk,
```

Comment:
@jdstrand the snap in question is contained in the [snapd codebase](https://github.com/snapcore/snapd/tree/master/tests/lib/snaps/snapctl-hooks). You can see the `configure` hook doesn't do anything other than run `snapctl` a few times. Note also that this works fine if the snap is installed-- the problems come when the snap is being _tried_. I assume that's where the access to my $HOME is coming from.